### ISSUE TITLE
Clicking connected dots will trigger onConnect event

### ISSUE THREAD

**Issue by pot-code** [wbkd/react-flow#559]
#  🐛 Bug Report

As title says.

console output:

```
Warning: Encountered two children with the same key, `reactflow__edge-1-1`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
    in g (created by EdgeRenderer)
    in svg (created by EdgeRenderer)
    in EdgeRenderer (created by GraphView)
    in div (created by FlowRenderer)
    in FlowRenderer (created by GraphView)
    in GraphView (created by ReactFlow)
    in StoreProvider (created by ReactFlowWrapper)
    in ReactFlowWrapper (created by ReactFlow)
    in div (created by ReactFlow)
    in ReactFlow (at App.js:49)
    in div (created by Context.Consumer)
    in EmotionCssPropInternal (at App.js:38)
    in App (at src/index.js:9)
    in StrictMode (at src/index.js:8)
```

## To Reproduce

1. connect two nodes
2. click any dot on one of the nodes

## Expected behavior

Should not trigger onConnect event

## Link to repl or repo

https://codesandbox.io/s/react-flow559-ddxhf

## envinfo

package version: 6.1.1
OS: 64bit Mac OS X 10.14.6 18G6020

**Comment by chrtze** [comment]
Hi, I think this is the expected behaviour. If you want to prevent nodes being connected to themselves, you can do a check in your `onConnect`, something like `params.source !== params.target`. Anyway, we should take a look into the duplicate key warning separately.

**Comment by pot-code** [comment]
> Hi, I think this is the expected behaviour. If you want to prevent nodes being connected to themselves, you can do a check in your `onConnect`, something like `params.source !== params.target`. Anyway, we should take a look into the duplicate key warning separately.

@chrtze Hi, thank for your reply, but I still think there is a misunderstanding. IT IS a CLICK action, not CONNECT, thus it would be weird to get a onConnect event.

**Comment by chrtze** [comment]
You are right, in that case we should think about adding a threshold for checking if the user has dragged. Thanks for clarifying!

**Comment by code-guru** [comment]
I had the same issue, in my use case, a node could not connect to itself, so preventing the connection solved the issue. But I also agree clicking on handle should not trigger connect.

**Comment by moklick** [comment]
I could image that this will be fixed when we merge #543. Otherwise we can add a flag `hasMoved` in order to check if user has moved the mouse or just clicked.

**Comment by chrtze** [comment]
This will be fixed in the next version. You can already try it out with `npm install react-flow-renderer@next`. #657

**Comment by chrtze** [comment]
released in [7.0.0](https://github.com/wbkd/react-flow/releases/tag/7.0.0)
